### PR TITLE
Remove unused frontend view features

### PR DESCRIPTION
**Pull request by mbj**
* We do have more well specified and correct equivalents in our main app
* So carrying the code weight and the specs for those features around
  here is wasteful

Extraction from #398

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review.

File: frontend/app/views/spree/address/_form.html.erb [hunk 1]
```diff
@@ -33,30 +33,10 @@
...
-      %>
-      <%= javascript_tag do -%>
-        $('#<%="#{address_id}state" %>').append("<%== state_elements %>");
-      <% end %>
-    </p>
-      <noscript>
-        <%= form.text_field :state_name, :class => 'required' %>
-      </noscript>
-  <% end %>
+  <p class="field" id=<%="#{address_id}state" %>>
```
Comment (dkubb) [inline frontend/app/views/spree/address/_form.html.erb]: What about moving the string outside of the erb, and writing this as:

```erb
<p class="field" id="<%= address_id %>state">
```

This has the added benefit of correcting the invalid, unquoted HTML attribute.

File: frontend/app/views/spree/address/_form.html.erb [hunk 2]
```diff
@@ -33,30 +33,10 @@
...
+    <%= form.label :state, Spree.t(:state) %>
+    <%= form.collection_select :state_id, address.country.try(:states) || [], :id, :name %>
```
Comment (dkubb) [inline frontend/app/views/spree/address/_form.html.erb]: What about: `address.country&.states || []`

File: frontend/app/views/spree/address/_form.html.erb (hunk 1 above)
Comment (mbj) [inline frontend/app/views/spree/address/_form.html.erb]: we do not use this code at all its only there to drive the headless specs. I only included the `<p` tag for consistency (with exactly that interpolation style) with the other elements in this view.

Comment (dkubb) [inline frontend/app/views/spree/address/_form.html.erb]: I'd probably still make the change. You had to touch the code this time, so if there's a next time you want the code to be as simple to understand as possible.

Comment (mbj) [inline frontend/app/views/spree/address/_form.html.erb]: I'm arguing against this change as I had to fix lots of other exactly similar (but data) `<p>` tags in this file for consistency.

Comment (dkubb) [inline frontend/app/views/spree/address/_form.html.erb]: I must've missed those. Was that in another PR?

Comment (mbj) [inline frontend/app/views/spree/address/_form.html.erb]: this file contains 22 `<p class="field" id=<%="#{address_id}....>` adjacent to this change. I'd like to keep these in-line for now and focus on the intend of this change, rather to increase to a global style fix.

Comment (dkubb) [inline frontend/app/views/spree/address/_form.html.erb]: Oh, I see. If there are others like this in the file you'd have to change for consistency, then I see why you don't want to do it.

Comment (mbj) [inline frontend/app/views/spree/address/_form.html.erb]: > Oh, I see. If there are others like this in the file you'd have to change for consistency, then I see why you don't want to do it.

Yeah, sorry for not being clear initially.

**Comment by dkubb** [comment]
@mbj I am fine with this change. Will merge on green.

**Pull request merged**